src/impression.js

```diff
@@ -47,15 +53,72 @@ export function maybeTrackImpression(win) {
     // avoid duplicate tracking.
     win.location.hash = '';
   }
+
   viewer.whenFirstVisible().then(() => {
-    invoke(win, clickUrl);
+    // TODO(@zhouyx) We need a timeout here?
+    // TODO(@zhouyx) need test with a real response.
+    invoke(win, clickUrl).then(response => {
+      if (!response) {
+        return;
+      }
+      applyResponse(win, viewer, response);
+    });
   });
 }
 
+/**
+ * Send the url to ad server and wait for its response
+ * @param {!Window} win
+ * @param {string} clickUrl
+ * @return {!Promise}
+ */
 function invoke(win, clickUrl) {
-  xhrFor(win).fetchJson(clickUrl, {
+  if (getMode().localDev && !getMode().test) {
+    clickUrl = 'http://localhost:8000/impression-proxy?url=' + clickUrl;
+  }
+  return xhrFor(win).fetchJson(clickUrl, {
     credentials: 'include',
     requireAmpResponseSourceOrigin: true,
   });
   // TODO(@cramforce): Do something with the result.
 }
+
+/**
+ * parse the response back from ad server
+ * Set for analytics purposes
+ * @param {!Window} win
+ * @param {!Object} response
+ */
+function applyResponse(win, viewer, response) {
+  const adLocation = response['location'];
+
+  // If there's a tracking_url, need to redirect to that url.
+  const adTrackingUrl = response['tracking_url'];
+  if (adTrackingUrl) {
+    // TODO(@zhouyx) Confirm we assume the tracking_url is not
+    // a cdn.ampproject page, and we don't do viewer redirect
+    openWindowDialog(win, adTrackingUrl, '_top');
```

Comment:
Not sure where this came frame. Just request the URL with `new Image().src = adTrackingUrl`